packages/backend-storage/src/policy_arbiter.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  ConstructFactoryGetInstanceProps,
+  ResourceAccessAcceptor,
+  SsmEnvironmentEntriesGenerator,
+} from '@aws-amplify/plugin-types';
+import { AccessGenerator } from './types.js';
+import { IBucket } from 'aws-cdk-lib/aws-s3';
+import { storageAccessBuilder } from './access_builder.js';
+import { BucketPolicyFactory, Permission } from './policy_factory.js';
+
+/**
+ * Middleman between creating bucket policies and attaching those policies to corresponding roles
+ */
+export class BucketPolicyArbiter {
+  /**
+   * Instantiate with context from the storage factory
+   */
+  constructor(
+    private readonly name: string,
+    private readonly accessGenerator: AccessGenerator,
+    private readonly ssmEnvironmentEntriesGenerator: SsmEnvironmentEntriesGenerator,
+    private readonly getInstanceProps: ConstructFactoryGetInstanceProps,
+    private readonly bucket: IBucket,
+    private readonly bucketPolicyFactory = new BucketPolicyFactory(bucket)
+  ) {}
+
+  /**
+   * Responsible for evaluating the access definition,
+   * creating bucket policies corresponding to the definition,
+   * then invoking the corresponding resource access acceptor to accept the policies
+   */
+  arbitratePolicies = () => {
+    const accessDefinition = this.accessGenerator(storageAccessBuilder);
+
+    const accessMap: Map<ResourceAccessAcceptor, Permission[]> = new Map();
+
+    Object.entries(accessDefinition).forEach(
+      ([s3Prefix, accessPermissions]) => {
+        accessPermissions.forEach((permission) => {
+          const resourceAccessAcceptor = permission.getResourceAccessAcceptor(
+            this.getInstanceProps
+          );
+          if (!accessMap.has(resourceAccessAcceptor)) {
+            accessMap.set(resourceAccessAcceptor, []);
+          }
+          const prefix = s3Prefix.replaceAll(
```

Comment:
This will be iterated on in future PRs, but the rough DX being proposed at this point is  specifying paths like `/something/{owner}/*` and then we replace `{owner}` with the cognito identity id token in the policy